File: python/단어변환.py
def dfs(begin, target, words, w, s):
    if begin == target:
        return len(words)-len(w)
    for i in range(len(w)):
        wrong = 0
        for j in range(len(begin)):
            if begin[j] != w[i][j]:
                wrong += 1
            if wrong > 1:
                break
        if wrong == 1:
            s = dfs(w[i], target, words, w[:i]+w[i+1:], s+1)
    return s

def solution(begin, target, words):
    
    if target not in words:
        return 0

    answer = 50
    answer = min(answer, dfs(begin, target, words, words, 0))

    return answer

solution("hit", "cog", ["hot", "dot", "dog", "lot", "log", "cog"])

File: JavaScript/단어변환.js
function solution(begin, target, words) {
    if (!words.includes(target)) return 0;
    return dfs(begin, target, words, []);
}

function dfs(begin, target, words, visited) {
    if (begin === target) return visited.length;
    
    let difference, c = 51;
    words.forEach((word) => {
        if (!visited.includes(word)) {
            difference = 0;
            for (let i = 0; i < word.length; i++) if (begin[i] !== word[i]) difference++;
            if (difference === 1) {
                let v = visited.slice();
                v.push(word);
                let c_temp = dfs(word, target, words, v);
                if (c_temp != undefined) c = Math.min(c, c_temp);
            }
        }
    })
    return c;
}

console.log(solution("hit", "cog", ["hot", "dot", "dog", "lot", "log", "cog"]));